src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/UpsertResource_7.sql

```diff
@@ -119,50 +122,24 @@ FROM   dbo.Resource WITH (UPDLOCK, HOLDLOCK)
 WHERE  ResourceTypeId = @resourceTypeId
        AND ResourceId = @resourceId
        AND IsHistory = 0;
-IF (@etag IS NOT NULL
-    AND @etag <> @previousVersion)
-    BEGIN
-        THROW 50412, 'Precondition failed', 1;
-    END
+
 DECLARE @version AS INT; -- the version of the resource being written
 IF (@previousResourceSurrogateId IS NULL)
     BEGIN
-        
-		-- There is no previous version of this resource
-		IF (@isDeleted = 1)
-            BEGIN
-                -- Don't bother marking the resource as deleted since it already does not exist.
-				COMMIT TRANSACTION;
-                RETURN;
-            END
-        IF (@etag IS NOT NULL)
-            BEGIN
-                -- You can't update a resource with a specified version if the resource does not exist
-				THROW 50404, 'Resource with specified version not found', 1;
-            END
-        IF (@allowCreate = 0)
-            BEGIN
-                THROW 50405, 'Resource does not exist and create is not allowed', 1;
-            END
+        -- There is no previous version
         SET @version = 1;
     END
 ELSE
     BEGIN
         -- There is a previous version
-        IF (@requireETagOnUpdate = 1
-            AND @etag IS NULL)
+        -- When not a delete and if @comparedVersion deosn't matche the @previousVersion in the DB
+        IF (@isDeleted = 0 AND @comparedVersion <> @previousVersion) -- When not a delete
```

Comment:
>@comparedVersion <> @previousVersi [](http://example.com/codeflow?start=31&length=34)

:-) This is not C#
DECLARE @comparedVersion int = NULL, @previousVersion int = 1
SELECT CASE WHEN @comparedVersion <> @previousVersion THEN 1 ELSE 0 END
0

Reply:
Aha, you are right how about [this](https://github.com/microsoft/fhir-server/pull/2519/commits/ab568485d6a453ba96670b94fe3935ef848f76f2)?

Reply:
DECLARE @comparedVersion int = NULL, @previousVersion int = 1
SELECT CASE WHEN @comparedVersion IS NULL OR @comparedVersion <> @previousVersion THEN 1 ELSE 0 END
1